rust-app/src/parser/object.rs

```diff
@@ -0,0 +1,435 @@
+use crate::parser::common::*;
+use arrayvec::ArrayVec;
+use core::convert::TryInto;
+use core::future::Future;
+#[cfg(feature = "speculos")]
+use ledger_crypto_helpers::common::HexSlice;
+use ledger_crypto_helpers::hasher::{Blake2b, Hasher, HexHash};
+use ledger_device_sdk::io::SyscallError;
+use ledger_log::info;
+use ledger_parser_combinators::async_parser::*;
+use ledger_parser_combinators::bcs::async_parser::*;
+use ledger_parser_combinators::core_parsers::*;
+use ledger_parser_combinators::interp::*;
+
+// Object Schema
+pub type ObjectInnerSchema = (
+    ObjectDataSchema,
+    OwnerSchema,
+    TransactionDigest,
+    StorageRebate,
+);
+
+pub type MoveObject = (MoveObjectType, SequenceNumber, ObjectContents);
+
+// The object content parsing is limited to either a simple Coin (40 bytes) or StakedIota (80 bytes)
+// We will simply reject parsing objects with content size greater than OBJECT_CONTENTS_LEN
+pub const OBJECT_CONTENTS_LEN: usize = 80;
+pub type ObjectContents = Vec<Byte, OBJECT_CONTENTS_LEN>;
+pub type Coin = (UID, Amount);
+
+pub struct ObjectDataSchema;
+pub struct OwnerSchema;
+
+pub type TransactionDigest = Sha3_256Hash;
+pub type UID = ObjectID;
+
+pub type StorageRebate = U64LE;
+
+pub const STRING_LENGTH: usize = 64;
+pub type String = Vec<Byte, STRING_LENGTH>;
+
+pub type StructTag = (IotaAddress, String, String, TypeParams);
+pub type TypeParams = Vec<TypeTag2, 5>;
+
+pub struct TypeTag;
+
+// This is to avoid recursion of async parsers
+pub struct TypeTag2;
+
+// Parsed data
+pub enum MoveObjectType {
+    /// A type that is not `0x2::coin::Coin<T>`
+    // Other(StructTag),
```

Comment:
https://github.com/obsidiansystems/ledger-app-sui/commit/db01f6632ee7f9f099c6f684abc3f522724150ba
Any plan to enable this?